changelog.d/12858.bugfix

```diff
@@ -0,0 +1 @@
+Fix [MSC3878](https://github.com/matrix-org/matrix-spec-proposals/pull/3787) rooms being omitted from room directory, room summary and space hierarchy responses.
```

Comment:
MSC3787*

also, you're implying it's a bug and not an intentional feature :p 

(it's technically a bug, but was deliberately omitted from the original PR)

Reply:
I have no context; I seek only to make the tests on https://github.com/matrix-org/complement/pull/367 pass.

Reply:
Changelog updated in 6aeee9a19